include/cuco/detail/hyperloglog/tuning.cuh

```diff
@@ -0,0 +1,2577 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cuda/std/array>
+
+namespace cuco::hyperloglog_ns::detail {
+
+// TODO this will spawn one copy of each array in every TU :(
```

Comment:
How much device memory does this header end up using?

Reply:
So we don't need to include all of these sub-arrays into the TU, just the ones that correspond to the given `Precision` value. Long story short: in the worst case we need three arrays that together require <3.4KB.